src/cmd/services/m3query/config/config.go

```diff
@@ -146,9 +141,65 @@ type FilterConfiguration struct {
 	CompleteTags Filter `yaml:"completeTags"`
 }
 
-// LimitsConfiguration represents limitations on per-query resource usage. Zero or negative values imply no limit.
+// LimitsConfiguration represents limitations on resource usage in the query instance. Limits are split between per-query
+// and global limits.
 type LimitsConfiguration struct {
-	MaxComputedDatapoints int64 `yaml:"maxComputedDatapoints"`
+	// deprecated: use PerQuery.MaxComputedDatapoints instead.
+	DeprecatedMaxComputedDatapoints int64 `yaml:"maxComputedDatapoints"`
```

Comment:
Yeah that was my initial thought as well, but @richardartoul pointed out that any existing config files using this field won't load without the back compat layer, which would be a pretty bad time (e.g. folks upgrade their binary, and all of a sudden their server won't start). I think it might be the less of the two evils (though I'm definitely open to changing it if we come that consensus).

Reply:
Is this a necessary field in the config?

Reply:
"Necessary" in the sense that without it, any config files using `maxComputedDatapoints` will fail to parse, e.g.
```
# will hard error on startup
limits:
   maxComputedDatapoints: 5
```